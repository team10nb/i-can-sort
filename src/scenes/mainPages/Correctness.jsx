// The correctness main page, consists of algorithm modules, set and help buttons, and the choice menu
import React from "react";
import { Link } from "react-router-dom";
import Set from "../../components/Buttons/Set";
import Help from "../../components/Buttons/Help";
import Module from "../../components/Module/Module";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import logo1 from "../../Resource/Chelp1.png";
import logo2 from "../../Resource/Chelp2.png";
import logo3 from "../../Resource/Chelp3.png";
import { motion } from "framer-motion";
import img_tutorial from "../../Resource/tutorial.png";
import img_proof from "../../Resource/proof.png";

//Set css
const useStyles = makeStyles((theme) => ({
    root: {
        // minWidth: 200,
        display: "flex",
        "& > *": {
            marginLeft: 89,
        },
        marginLeft: -66,
    },
    button: {
        width: 230,
        fontFamily: "Quicksand",
        fontWeight: 1000,
    },
    grid: {
        flexGrow: 1,
        width: 900,
        height: 600,
        paddingLeft: 30,
        paddingTop: 30,
    },
    div: {
        textAlign: "center",
        height: 690,
        width: 960,
    },
    buttonSet: {
        display: "flex",
        "& > *": {
            margin: theme.spacing(0),
        },
    },
}));

//Set theme color
const color = "#ff6f00";

//Return a grid contains 8 modules that represent 8 sorting algorithms，set and help buttons，and the choice menu
//Click the modules to jump to the corresponding algorithm learning page
//The progress bar under each module represents the learning progress of the algorithm
export default function CorrectnessMainPage(props) {
    const classes = useStyles();
    //get progress info from local
    //  const localPre = localStorage.getItem("pre") ? JSON.parse(localStorage.getItem("pre")) : null;
    //  const localProgress = localStorage.getItem("progress") ? JSON.parse(localStorage.getItem("progress")) : 0;

    //set progress
    //  const progress = props.location.state ? props.location.state : localProgress;
    //  localStorage.setItem("progress", JSON.stringify(progress));

    //  const handleClick = (title) => () => {
    //    //store previous visited algorithm
    //    localStorage.setItem("pre", JSON.stringify(title));
    //  }

    //  const handlePre = (title) => {
    //    //set previous visited algorithm
    //    const preOne = (title === localPre) ? true : false;
    //    return preOne;
    //  }

    const images = [
        {
            static: img_tutorial,
            gif: "catch.gif",
            title: "Tutorial",
            width: "20%",
        },
        {
            static: img_proof,
            gif: "catch.gif",
            title: "Proof",
            width: "20%",
        },
    ];

    //   const TutorialProps = {
    //     image: images[0],
    //     width: 200,
    //     height: 200,
    //     onClick: handleClick(images[0].title),
    //     progress: progress,
    //     color: color,
    //     preOne: handlePre(images[0].title),
    // };

    const props1 = {
        image: images[1],
        width: 425,
        height: 435,
        // onClick: handleClick(images[3].title),
        progress: 0,
        color: color,
        // preOne: false,
    };

    const props2 = {
        image: images[0],
        width: 425,
        height: 435,
        // onClick: handleClick(images[3].title),
        progress: 0,
        color: color,
        // preOne: false,
    };

    const helpProp = {
        logo1: logo1,
        logo2: logo2,
        logo3: logo3,
        color: color,
    };

    return (
        <motion.div
            className={classes.div}
            initial={{ opacity: 0.2, x: "100vw" }}
            animate={{
                opacity: 1,
                scale: 1,
                x: "0vw",
            }}
            transition={{
                type: "spring",
                stiffness: 250,
                damping: 25,
            }}
            exit={{ opacity: 0.2, scale: 0, x: "100vw" }}
        >
            <div className={classes.grid}>
                <Grid container >
                    <Grid container item xs={12}  spacing={0}>
                    <Grid item xs={6} >
                        <Link to="/Correctness/Tutorial">
                            <Module {...props2} />
                        </Link>
                    </Grid>

                    <Grid item xs={6} >
                        <Link to="/Correctness/Prove">
                            <Module {...props2} />
                        </Link>
                    </Grid>
                    </Grid>

                    <Grid item xs={12}>
                        <div className={classes.buttonSet}>
                            <Set />
                            <Help {...helpProp} />
                        </div>
                    </Grid>

                    <Grid item xs={12}>
                        <div style={{ marginLeft: -8, marginTop: 50 }}>
                            <CorrectnessChoiceMenu />
                        </div>
                    </Grid>
                </Grid>
            </div>
        </motion.div>
    );
}

//Return five button, which are 'Tutorial', 'Procedure', 'Efficiency', 'Correctness', and 'Exercise'
//The Correctness button is theme color without jump function
export function CorrectnessChoiceMenu() {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <Link to="/TutorialMainPage" style={{ textDecoration: "none" }}>
                <Button variant="contained" className={classes.button}>
                    Tutorial
                </Button>
            </Link>

            <Link to="/ProcedureMainPage" style={{ textDecoration: "none" }}>
                <Button variant="contained" className={classes.button}>
                    Procedure
                </Button>
            </Link>

            <Button
                variant="contained"
                color="primary"
                disableElevation
                className={classes.button}
                style={{ color: "white", backgroundColor: color }}
            >
                Correctness
            </Button>
        </div>
    );
}

export { color };
