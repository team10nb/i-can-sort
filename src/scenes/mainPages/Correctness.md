
### **Developer**
Yijie Lu

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

This is a correctness main page. It contains correctness's modules.

###  **Import**

```html
import { makeStyles } from '@material-ui/core/styles';
```

###  **Components**

| Name | Description | Child components |
| ---- | ----------- | ---------------- |
| Module |    This button contains an image and a gif to indicate the sorting process. Gif will be shown when the cursor hovers.    |       \           |   
| Setting Button |    It is a set button. By clicking it, users can change language, background music and see some information about the team.      |      \       |
| Help Button |    It is a help button, after clicking it, the explanation of each interface's component will be shown.      |       \        |
| Section Navigation  |    a navigation used to switch different pages    |       \        | 

###  **SubPages**

| Name   | Description | 
| ----   | ----------- | 
| Correctness Tutorial | This is a correctness tutorial. Users can get information about correctness from here.|
| Correctness Proof | This is a correctness proof. Users can get information about correctness's proof of eacch sorting algortihm.|


