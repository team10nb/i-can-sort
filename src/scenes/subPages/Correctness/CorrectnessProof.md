
### **Developer**
Yijie Lu, Shiliang Chen

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

This is a correctness proof. Users can get information about correctness's proof of eacch sorting algortihm.

###  **Import**

```html
import MobileStepper from "@material-ui/core/MobileStepper";
import Button from "@material-ui/core/Button";
import Drawer from "@material-ui/core/Drawer";
import AppBar from "@material-ui/core/AppBar";
import Chip from "@material-ui/core/Chip";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";
import HomeIcon from "@material-ui/icons/Home";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";
import { BubbleProof, SelectionProof, InsertionProof, QuickProof, HeapProof, MergeProof} from "src/components/CorrectnessProof/CorrectnessProof";
import ProofHelp from "src/components/CorrectnessProof/ProofHelp";
```

###  **Components**

| Name | Description | Child components|
| ---- | ----------- | ----------------|
| AppBar1 |    a bar contains each subpage's name     |        |   
|Drawer |    a drawer show the level of subpage     |       List        |
| List   |    a list of each subpage's name         |            |
| AppBar2 |   a bar show page number and give access to the last and the next page|    |   

###  **SubPages**

| Name   | Description | 
| ----   | ----------- | 
| Bubble | Bubble's proof|
| Selection | Selection's proof|
| Insertion | Insertion's proof|
| Quick | Quick's proof|
| Merge | Merge's proof|
| Heap | Heap's proof|

