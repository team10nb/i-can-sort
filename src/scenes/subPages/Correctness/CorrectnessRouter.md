
### **Developer**
Yijie Lu, Yuting Jiang

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**
create router of each subpage.

###  **Import**

```html
import { makeStyles } from '@material-ui/core/styles';
```

###  **Props**

| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
|CorrectnessRouters|path: string;  component: (props: any) => JSX.Element;||create router of each each subpage|


###  **Methods**



###  **Demo**

```jsx

```