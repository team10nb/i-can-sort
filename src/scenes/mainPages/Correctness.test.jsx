import React from 'react';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router'
import CorrectnessMainPage from './Correctness';
import { color, CorrectnessChoiceMenu } from './Correctness';

test('should export theme color', () =>{
    const testColor = color;
    const CorrectnessThemeColor = '#ff6f00';
    expect(testColor).toBe(CorrectnessThemeColor);
})

test('should contain CorrectnessChoiceMenu', () => {
    const {getByText} = render(
        <StaticRouter>
            <CorrectnessChoiceMenu />
        </StaticRouter>
    );
    getByText("Tutorial");
    getByText("Procedure");
    getByText("Efficiency");
    getByText("Correctness");
    getByText("Exercise");
})

test('should match Snapshot', () => {
    const props = { location: "/", }
    const content = render(
        <StaticRouter>
            <CorrectnessMainPage {...props}/>
        </StaticRouter>
    );
    expect(content).toMatchSnapshot();
})