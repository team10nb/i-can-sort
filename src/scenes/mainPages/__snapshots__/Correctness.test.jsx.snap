// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-div-9"
        style="opacity: 0.2; transform: translateX(100vw) translateZ(0);"
      >
        <div
          class="makeStyles-grid-8"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <a
                  href="/Correctness/Tutorial"
                >
                  <div
                    style="display: inline-block; margin: 10px;"
                  >
                    <div
                      class="makeStyles-root-11"
                    >
                      <button
                        class="MuiButtonBase-root makeStyles-image-12"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="makeStyles-imageSrc-15"
                          style="background-color: rgb(213, 213, 213);"
                        />
                        <span
                          class="makeStyles-imageBackdrop-16"
                        />
                        <span
                          class="makeStyles-imageButton-14"
                        >
                          <span
                            class="MuiTypography-root makeStyles-imageTitle-17 MuiTypography-subtitle1 MuiTypography-colorInherit"
                          >
                            Tutorial
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="makeStyles-root-18"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="0"
                        class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-19 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-20 MuiLinearProgress-determinate"
                        role="progressbar"
                      >
                        <div
                          class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-21 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                          style="transform: translateX(-100%);"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <a
                  href="/Page"
                >
                  <div
                    style="display: inline-block; margin: 10px;"
                  >
                    <div
                      class="makeStyles-root-22"
                    >
                      <button
                        class="MuiButtonBase-root makeStyles-image-23"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="makeStyles-imageSrc-26"
                          style="background-color: rgb(213, 213, 213);"
                        />
                        <span
                          class="makeStyles-imageBackdrop-27"
                        />
                        <span
                          class="makeStyles-imageButton-25"
                        >
                          <span
                            class="MuiTypography-root makeStyles-imageTitle-28 MuiTypography-subtitle1 MuiTypography-colorInherit"
                          >
                            Proof
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="makeStyles-root-29"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="0"
                        class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-30 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-31 MuiLinearProgress-determinate"
                        role="progressbar"
                      >
                        <div
                          class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-32 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                          style="transform: translateX(-100%);"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="makeStyles-buttonSet-10"
              >
                <div>
                  <button
                    aria-label="info"
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    title="information"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                       
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    class="PrivateSwipeArea-root-41 PrivateSwipeArea-anchorLeft-42"
                    style="width: 20px;"
                  />
                </div>
                <div>
                  <button
                    aria-label="help"
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    title="tutorial"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    aria-hidden="true"
                    class="MuiBackdrop-root makeStyles-backdrop-36"
                    style="opacity: 0; visibility: hidden;"
                  >
                    <div
                      class="MuiPaper-root MuiCard-root makeStyles-card-37 MuiPaper-elevation1 MuiPaper-rounded"
                    >
                      <div
                        class="makeStyles-avatar-38"
                      >
                        <button
                          aria-label="close"
                          class="MuiButtonBase-root MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              style="color: rgb(255, 111, 0); opacity: 0.8; font-size: 40px;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <div>
                          <button
                            class="MuiButtonBase-root MuiCardActionArea-root"
                            style="width: 580px; height: 403px;"
                            tabindex="0"
                            type="button"
                          >
                            <img
                              alt="logo1"
                              heigh="300"
                              src="Chelp1.png"
                              width="587"
                            />
                            <div
                              class="MuiCardContent-root"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                              >
                                This a entry for each sorting algorithm. Choose one and click it to enter next Page. A progress bar is at the bottom, it shows how much you have learnt.
                              </p>
                            </div>
                            <span
                              class="MuiCardActionArea-focusHighlight"
                            />
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <div
                          class="makeStyles-button-39"
                        >
                          <button
                            class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                            disabled=""
                            style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                style="color: white; font-size: 40px;"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                                />
                              </svg>
                            </span>
                          </button>
                          <button
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                            style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                style="color: white; font-size: 40px;"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                style="margin-left: -8px; margin-top: 50px;"
              >
                <div
                  class="makeStyles-root-6"
                >
                  <a
                    href="/TutorialMainPage"
                    style="text-decoration: none;"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Tutorial
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                  <a
                    href="/ProcedureMainPage"
                    style="text-decoration: none;"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Procedure
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary MuiButton-disableElevation"
                    style="color: white; background-color: rgb(255, 111, 0);"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Correctness
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-div-9"
      style="opacity: 0.2; transform: translateX(100vw) translateZ(0);"
    >
      <div
        class="makeStyles-grid-8"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <a
                href="/Correctness/Tutorial"
              >
                <div
                  style="display: inline-block; margin: 10px;"
                >
                  <div
                    class="makeStyles-root-11"
                  >
                    <button
                      class="MuiButtonBase-root makeStyles-image-12"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="makeStyles-imageSrc-15"
                        style="background-color: rgb(213, 213, 213);"
                      />
                      <span
                        class="makeStyles-imageBackdrop-16"
                      />
                      <span
                        class="makeStyles-imageButton-14"
                      >
                        <span
                          class="MuiTypography-root makeStyles-imageTitle-17 MuiTypography-subtitle1 MuiTypography-colorInherit"
                        >
                          Tutorial
                        </span>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="makeStyles-root-18"
                  >
                    <div
                      aria-valuemax="100"
                      aria-valuemin="0"
                      aria-valuenow="0"
                      class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-19 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-20 MuiLinearProgress-determinate"
                      role="progressbar"
                    >
                      <div
                        class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-21 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                        style="transform: translateX(-100%);"
                      />
                    </div>
                  </div>
                </div>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <a
                href="/Page"
              >
                <div
                  style="display: inline-block; margin: 10px;"
                >
                  <div
                    class="makeStyles-root-22"
                  >
                    <button
                      class="MuiButtonBase-root makeStyles-image-23"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="makeStyles-imageSrc-26"
                        style="background-color: rgb(213, 213, 213);"
                      />
                      <span
                        class="makeStyles-imageBackdrop-27"
                      />
                      <span
                        class="makeStyles-imageButton-25"
                      >
                        <span
                          class="MuiTypography-root makeStyles-imageTitle-28 MuiTypography-subtitle1 MuiTypography-colorInherit"
                        >
                          Proof
                        </span>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="makeStyles-root-29"
                  >
                    <div
                      aria-valuemax="100"
                      aria-valuemin="0"
                      aria-valuenow="0"
                      class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-30 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-31 MuiLinearProgress-determinate"
                      role="progressbar"
                    >
                      <div
                        class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-32 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                        style="transform: translateX(-100%);"
                      />
                    </div>
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="makeStyles-buttonSet-10"
            >
              <div>
                <button
                  aria-label="info"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  title="information"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                     
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="PrivateSwipeArea-root-41 PrivateSwipeArea-anchorLeft-42"
                  style="width: 20px;"
                />
              </div>
              <div>
                <button
                  aria-label="help"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  title="tutorial"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  aria-hidden="true"
                  class="MuiBackdrop-root makeStyles-backdrop-36"
                  style="opacity: 0; visibility: hidden;"
                >
                  <div
                    class="MuiPaper-root MuiCard-root makeStyles-card-37 MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="makeStyles-avatar-38"
                    >
                      <button
                        aria-label="close"
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="color: rgb(255, 111, 0); opacity: 0.8; font-size: 40px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div>
                      <div>
                        <button
                          class="MuiButtonBase-root MuiCardActionArea-root"
                          style="width: 580px; height: 403px;"
                          tabindex="0"
                          type="button"
                        >
                          <img
                            alt="logo1"
                            heigh="300"
                            src="Chelp1.png"
                            width="587"
                          />
                          <div
                            class="MuiCardContent-root"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                            >
                              This a entry for each sorting algorithm. Choose one and click it to enter next Page. A progress bar is at the bottom, it shows how much you have learnt.
                            </p>
                          </div>
                          <span
                            class="MuiCardActionArea-focusHighlight"
                          />
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div
                        class="makeStyles-button-39"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                          disabled=""
                          style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              style="color: white; font-size: 40px;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                              />
                            </svg>
                          </span>
                        </button>
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          style="background-color: rgb(255, 111, 0); opacity: 0.5;"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              style="color: white; font-size: 40px;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              style="margin-left: -8px; margin-top: 50px;"
            >
              <div
                class="makeStyles-root-6"
              >
                <a
                  href="/TutorialMainPage"
                  style="text-decoration: none;"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Tutorial
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  href="/ProcedureMainPage"
                  style="text-decoration: none;"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Procedure
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary MuiButton-disableElevation"
                  style="color: white; background-color: rgb(255, 111, 0);"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Correctness
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
